slides/week5/git.org

```diff
@@ -0,0 +1,37 @@
+#+TITLE: Git Training - Week 5
+#+AUTHOR: Sahit Chintalapudi, Josh Ting
+#+EMAIL: schintalapudi@gatech.edu, josh.ting@gatech.edu
+
+* Merge Conflicts
+- Two weeks ago, we talked about branches and how they can be used to avoid
+  conflicts in the code
+- But sometimes, multiple people edit the same lines of code on the same branch. Git's
+  powerful, but it's not magic.
+- A merge conflict occurs when you try to push code based on an old version
+  of the software that conflicts with changes in the same files already
+  pushed.
+  [[file:https://developer.atlassian.com/blog/2015/01/a-better-pull-request/merge-conflict.png]]
+
+* What does a merge conflict look like
+#+BEGIN_SRC shell
+some source code...
+<<<<<< HEAD
+This is a change I made
+==========
+This is a different change someone else made in the same place
+>>>>>> The hash of the other commit, something like fa54dd6f3...
+The rest of the source that had not been touched
+#+END_SRC
+- When you pull changes that lead to a merge conflict Git will notify which
+  files have a merge conflict in them
+- In each file, Git inserts the above text to indicate a merge conflict
+
+* So how do we deal with this?
+- One strategy is to just go through the file and manually reconcile the
+  conflict
+- After compiling and testing your changes to make sure the new code works, you
+  have to commit your merge resolution before being able to push your
+  changes.
+
+* KDiff3
+[[file:https://i.imgur.com/LkXr9sa.png]]
```

Comment:
kdiff seems unmaintained. I would just reccomend removing this, since gui merge managers are often more confusing and easier to screw things up in than the normal method.